src/ci_provider/buildkite/provider.rs

```diff
@@ -0,0 +1,336 @@
+    )
+    .expect("Failed to compile GitHub URL regex");
+}
+
+impl BuildkiteProvider {
+    /// Returns the pull request number if it is a pull request, otherwise returns None.
+    pub fn get_pr_number() -> Result<Option<u64>> {
+        Ok(get_env_variable("BUILDKITE_PULL_REQUEST")?.parse().ok())
+    }
+
+    pub fn get_run_event() -> Result<RunEvent> {
+        let is_pr = Self::get_pr_number()?.is_some();
+
+        if is_pr {
+            Ok(RunEvent::PullRequest)
+        } else {
+            Ok(RunEvent::Push)
+        }
+    }
+
+    pub fn get_ref() -> Result<String> {
+        let pr_number = Self::get_pr_number()?;
+
+        if let Some(pr_number) = pr_number {
+            Ok(format!("refs/pull/{}/merge", pr_number))
+        } else {
+            Ok(format!(
+                "refs/heads/{}",
+                get_env_variable("BUILDKITE_BRANCH")?
+            ))
+        }
+    }
+
+    pub fn get_owner_and_repository() -> Result<(String, String)> {
+        let repository_url = get_env_variable("BUILDKITE_REPO")?;
+        let captures = GITHUB_URL_REGEX
+            .captures(&repository_url)
+            .context("Failed to parse the GitHub repository URL")?;
+
+        let owner = captures
+            .name("owner")
+            .context("Failed to parse the GitHub repository URL")?
+            .as_str();
+        let repository = captures
+            .name("repository")
+            .context("Failed to parse the GitHub repository URL")?
+            .as_str();
+
+        Ok((owner.into(), repository.into()))
+    }
+}
+
+impl TryFrom<&Config> for BuildkiteProvider {
+    type Error = Error;
+    fn try_from(config: &Config) -> Result<Self> {
+        if config.token.is_none() {
+            bail!("Token authentication is required for Buildkite");
+        }
+
+        let is_pr = Self::get_pr_number()?.is_some();
+        let (owner, repository) = Self::get_owner_and_repository()?;
+
+        Ok(Self {
+            owner: owner.clone(),
+            repository: repository.clone(),
+            ref_: Self::get_ref()?,
+            base_ref: if is_pr {
+                Some(get_env_variable("BUILDKITE_PULL_REQUEST_BASE_BRANCH")?)
+            } else {
+                None
+            },
+            head_ref: if is_pr {
+                Some(get_env_variable("BUILDKITE_BRANCH")?)
+            } else {
+                None
+            },
+            commit_hash: get_env_variable("BUILDKITE_COMMIT")?,
+            event: Self::get_run_event()?,
+            repository_root_path: format!(
+                "/buildkite/builds/{}/{}/{}/",
+                get_env_variable("BUILDKITE_AGENT_NAME")?,
+                get_env_variable("BUILDKITE_ORGANIZATION_SLUG")?,
+                get_env_variable("BUILDKITE_PIPELINE_SLUG")?,
+            ),
+        })
+    }
+}
+
+impl CIProviderDetector for BuildkiteProvider {
+    fn detect() -> bool {
+        // check if the BUILDKITE environment variable is set and the value is truthy
```

Comment:
```suggestion
```

Reply:
No comment needed imo and it's not exactly checking if it's truthy (`1` would not pass)